tests/10-create.js

```diff
@@ -40,20 +43,92 @@ describe('eddsa-2022 (create)', function() {
+            verificationMethodDocument =>
+              verificationMethodDocument?.type === 'Multikey'
+          ).should.equal(true, 'Expected at least one proof to have "type" ' +
+            'property value "Multikey".');
         });
-        it('The field "cryptosuite" MUST be `eddsa-2022`', function() {
+        it('The "proof.proofPurpose" field MUST match the verification ' +
+          'relationship expressed by the verification method controller.',
+        async function() {
           this.test.cell = {columnId, rowId: this.test.title};
+          verificationMethodDocuments.should.not.eql([], 'Expected ' +
+            'at least one "verificationMethodDocument".');
+          verificationMethodDocuments.some(
+            verificationMethodDocument =>
+              verificationMethodDocument?.type === 'Multikey'
+          ).should.equal(true, 'Expected at least one proof to have "type" ' +
+            'property value "Multikey".'
+          );
+          const controllerDocuments = [];
+          for(const verificationMethodDocument of verificationMethodDocuments) {
+            const controllerDocument = await documentLoader({
+              url: verificationMethodDocument.controller
+            });
+            controllerDocuments.push(controllerDocument);
+          }
           proofs.some(
-            proof => proof?.cryptosuite === cryptosuite
-          ).should.equal(
-            true,
-            'Expected at least one proof to have "cryptosuite" `eddsa-2022`.'
+            proof => controllerDocuments.some(controllerDocument =>
+              controllerDocument.hasOwnProperty(proof.proofPurpose))
+          ).should.equal(true, 'Expected "proof.proofPurpose" field ' +
+             'to match the verification method controller.'
           );
         });
+        it('The "publicKeyMultibase" value of the verification method MUST ' +
+          'be 35 bytes in length and starts with the base-58-btc prefix (z).',
+        async function() {
+          this.test.cell = {columnId, rowId: this.test.title};
+          verificationMethodDocuments.should.not.eql([], 'Expected ' +
+            'at least one "verificationMethodDocument".');
+          for(const verificationMethodDocument of verificationMethodDocuments) {
+            const multibase = 'z';
+            const {publicKeyMultibase} = verificationMethodDocument;
+            const isMultibaseEncoded =
+              publicKeyMultibase.startsWith(multibase) &&
+                shouldBeBs58(publicKeyMultibase);
+            isMultibaseEncoded.should.equal(
+              true,
+              'Expected "publicKeyMultibase" value of the verification ' +
+              'method to be multibase base58-btc encoded value'
+            );
+            const publicKeyMultibaseBytes = bs58Decode({
+              id: publicKeyMultibase
+            });
+            publicKeyMultibaseBytes.byteLength.should.equal(35, 'Expected ' +
```

Comment:
I'm not sure if the assertion here is correct. The spec states that the byte length of the `publicKeyMultibase` should be 35 (see - https://www.w3.org/TR/vc-di-eddsa/#multikey). However, when the `publicKeyMulibase` is decoded, it appears to have a byte length of 34. Here's an example of what the publicKeyMultibase looks like: `z6Mkq6FxLL9UT6uLihkHRrjqnz7K9TDHhbTTH3bvJ7BbUgup`.

Is it possible that that is an error in the spec. What I noticed is that in the spec, it mentions a two-byte prefix (0xed01), which, when I encoded using varint, results in three bytes. Also, the prefix of "ed25519" in the multicodec table is 0xed (See -https://github.com/multiformats/multicodec/blob/master/table.csv#L97).

It would be great if someone could confirm if this is an error.

Reply:
@JSAssassin,

Someone requested that the varint text be deleted in this spec (and not the other, apparently), and this has led to some confusion. `0xed` as a varint is `0xed01`. So what's in the spec is already expressed as a varint -- so you don't need to convert it again. You can put just `0xed` into the varint API to produce the right value: `0xed01`, which is the value the spec says (it just doesn't mention varint anymore).

Reply:
Looks like the 35 is a copy-paste error from the ECDSA spec which has 35 byte key representations (as there's an extra "compression header" of a single byte there). It should be 34, so the tests should say that and we'll have to correct the typo in the spec.

Reply:
@dlongley Thanks a lot, I've fixed in the test as well as opened a PR to fix it in the spec https://github.com/w3c/vc-di-eddsa/pull/67